tests/mbed/enc/CMakeLists.txt

```diff
@@ -22,9 +22,23 @@ function(add_mbed_test_enclave NAME)
 
   set(MBEDTLS_TESTS_DIR "${PROJECT_SOURCE_DIR}/3rdparty/mbedtls/mbedtls/tests")
 
+# Use Perl that is found in the Path
+# If Perl is not found on Windows, use Perl from the Git/usr/bin directory
+# and report error on Linux
+  if (WIN32)
+    find_program(GIT git)
+    get_filename_component(GIT_DIR ${GIT} DIRECTORY)
+    set(PERL_HINTS "C:/Program Files/Git/usr/bin" "${GIT_DIR}/../usr/bin")
```

Comment:
Why is a hardcoded path being used here? 
Additionally it seems like a path separator is needed if two paths are to be specified.

Reply:
If user already has Perl installed, then we use that path. Otherwise, we look in /usr/bin under Git. The two specified hints cover all the possibilities.
Re: Path separator -- From CMake doc -- Multiple arguments to set will be joined as a ;-list to form the actual variable value to be set. i.e. a semicolon will be automatically added as separators to the multiple values.

Reply:
Why can we just not use git-bash's perl?

For example, the following is a case where we already use git-bash's sed:
https://github.com/Microsoft/openenclave/blob/master/3rdparty/musl/CMakeLists.txt#L72

COMMAND ${BASH} -c "sed -f ${MUSL_DIR}/tools/mkalltypes.sed ${MUSL_DIR}/arch/${ARCH}/bits/alltypes.h.in ${MUSL_DIR}/include/alltypes.h.in > ${MUSL_INCLUDES}/bits/alltypes.h"

Reply:
Yes, the two methods that worked were using Git Bash -c "perl ..."  and directly pointing to the Perl exe. But we chose to go with using Perl exe for reasons stated earlier. One solution for both Linux and Windows and also cleaner.

Reply:
I disagree that looking for perl.exe is cleaner. We already do platform specific lookup for BASH; so additionally we now look up perl as well using hard-coded paths.

1. There is no benefit to launching a custom perl rather than git bash's perl. What is the use case we are trying to support by looking up arbitrary perl installation?
2. There is no consistency with the new approach. Why is perl looked up this way and not sed as well?

Right now this solution feels ad-hoc; we prefer one way to do things for perl; another way to do things for sed.

Reply:
> we now look up perl as well using hard-coded paths.
This isn't quite true. This checks the PATH environment variable _and_ some extra paths as hints.

I do agree we should be consistent though; so if we kept this change then we should call sed directly, or we should use GIT_BASH in both cases.

I personally think GIT_BASH is _more_ hacky so I think that I'd prefer applying this approach to our use of sed too.

Reply:
I agree about making our code consistent, which can be done with another PR. The reason we see the non-conformance is because we conform to what the 3rd party code does to minimize upgrade time. The instances where Git bash -c "sed ..." is for the 3rd party CMakeLists.txt. We should invoke SED exe directly for these cases if we are ok on diverging from the 3rd party code base. But again, this calls for another PR.

This particular use of PERL_EXECUTABLE is in the tests code-base and also done in the mbedtls code-base.

For this PR, I like invoking PERL_EXECUTABLE directly (like the mbed tests and code do) without hard-coding the Paths but just providing hints to the find_program. We can ponder on making this consistent in our code-base with another PR.

The best approach, in my opinion, is to have the Windows users have Perl exe in their paths so that the logic to find Perl on Windows just goes away. This will require that we update our ACC Window VMs with Perl and our docs to reflect this new requirement. 

@anakrish - Please comment. Thanks.

Reply:
@anitagov 
I would like to keep the master simple and consistent; not have multiple ways to do the same thing.
I recommend that we follow the existing strategy ("bash -c") as part of this PR. 
If we think that the existing strategy is not correct; let us have another PR with an appropriate rationale.

Reply:
OK, @anakrish - The use of Perl is to generate/build the ELF enclaves aka the test cases. If we build the ELF enclaves on Linux, we are ok but for cross-compiling these on Windows, using Git Bash -c Perl may be keeping consistent from our end. Will make changes and send them out soon.

Reply:
@anakrish - Please review the changes to use Git Bash -c Perl method for Windows. **A sandbox CI run came out clean**. So, there are no issues at this time as I was concerned about the overhead of changing the Linux end of invoking ~50 bash shells to run perl.

Reply:
@anakrish - Please review the changes to use Git Bash -c Perl method for Windows. **A sandbox CI run came out clean**.